### PR TITLE
Vagrant based solution to create multi-node deployment

### PR DESCRIPTION
**Pull request by arvindkandhare**
Vagrant based solution to create multi-node deployment

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -51,7 +51,7 @@
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
     <zookeeper.version>3.5.1-alpha</zookeeper.version>
-    <bookkeeper.version>4.3.4-TWTTR</bookkeeper.version>
+    <bookkeeper.version>4.3.4-TWTTR-SNAPSHOT</bookkeeper.version>
```
Comment (franckcuny) [inline pom.xml]: is there a reason to use the snapshot instead of the release ?

File: vagrant/base.sh [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#!/bin/bash
+
+set -e
+
+if [ -z `which javac` ]; then
```
Comment (franckcuny) [inline vagrant/base.sh]: we try to follow Google's shell style. You should use `$(which javac)` instead of backticks here.

File: vagrant/base.sh [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    apt-get -y update
+    apt-get install -y software-properties-common python-software-properties
```
Comment (franckcuny) [inline vagrant/base.sh]: use `apt-get install -y --no-install-recommends` to avoid installing too many things

File: vagrant/zk.sh [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+log_dir=/opt/distributedlog-trunk/dist/release
+cd $log_dir
+
+cp $log_dir/distributedlog-service/conf/zookeeper.conf.template $log_dir/distributedlog-service/conf/zookeeper-$ZKID.properties
+echo "initLimit=5" >> $log_dir/distributedlog-service/conf/zookeeper-$ZKID.properties
+echo "syncLimit=2" >> $log_dir/distributedlog-service/conf/zookeeper-$ZKID.properties
+echo "quorumListenOnAllIPs=true" >> $log_dir/distributedlog-service/conf/zookeeper-$ZKID.properties
+sed  '/server.1/d' \
+    ./distributedlog-service/conf/bookie.conf.template > ./distributedlog-service/conf/zookeeper-$ZKID.conf
+for i in `seq 1 $NUM_ZK`; do
```
Comment (franckcuny) [inline vagrant/zk.sh]: for i in $(seq 1 $NUM_SK); do

**Comment by franckcuny** [comment]
Overall, this change looks good, I have a few minor suggestions and one question.

Thanks @arvindkandhare !

File: pom.xml (hunk 1 above)
Comment (arvindkandhare) [inline pom.xml]: This is to make sure that it takes this particular change: https://github.com/twitter/bookkeeper/pull/2 into account. But this works without this change too. Will update the pull request with all the comments taken care of.

Comment (arvindkandhare) [inline pom.xml]: Sorry, I realized that this works only with the said change request being pulled into BK. So we should get that in first.

Comment (sijie) [inline pom.xml]: @arvindkandhare I merged your bookkeeper pull request. will try to produce a version tomorrow

**Comment by franckcuny** [comment]
LGTM

Thanks!

File: pom.xml (hunk 1 above)
Comment (sijie) [inline pom.xml]: @arvindkandhare I pushed a version '4.3.5-TWTTR-OSS'. Could you update the version?

Comment (arvindkandhare) [inline pom.xml]: Sure. Updated the pull request with the latest version.

**Pull request merged**